src/libs/actions/Policy.js

```diff
@@ -949,7 +950,7 @@ function createWorkspace(ownerEmail = '', makeMeAdmin = false, policyName = '',
         ],
     });
 
-    Navigation.isNavigationReady()
+    return Navigation.isNavigationReady()
```

Comment:
Actions should not return a promise. Just revert this. The InteractionManager seems to be enough for our case, so no promise is actually needed.

Reply:
but didn't @roryabraham say so [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1682377415283179?thread_ts=1682369641.777619&cid=C01GTK53T8Q) ?

Reply:
This may create a pattern for further devs which I'm not sure if we want that.

But do we actually need a promise here?

Reply:
yes we do need it, as InteractionManager is not enough. And I think we were aware of this pattern change & accepted it before assigning/commencing work on this PR as per @roryabraham's comments in Slack.

Reply:
The promise used here does not wait for onyx. It waits for navigation, it just happen that at that point onyx data is updated.

Reply:
Actually at this point (_the following statement may change_) we are waiting for onyx data update since the optimistic data use onyx `SET` method which blocking (sync).

Reply:
> The promise used here does not wait for onyx. It waits for navigation, it just happen that at that point onyx data is updated.

Yes, that's a very good point. I think this is maybe fine for now, but long-term we do need a better way to wait for just the optimistic data to be written to cache and for subscribers to be updated.

> optimistic data use onyx SET method which is blocking (sync).

`Onyx.set` is async:

- https://github.com/Expensify/react-native-onyx/blob/0fbe0a4d077f96c4bc30b0b14359f634b271b03b/lib/Onyx.js#L862
- https://github.com/Expensify/react-native-onyx/blob/0fbe0a4d077f96c4bc30b0b14359f634b271b03b/lib/Onyx.js#L778

We could feasibly make a sync Onyx API on iOS and Android right now using JSI, but have no way to do so on web currently. We make `Onyx.set` async to make it more consistent with `Onyx.merge`

Reply:
@ntdiary If this is not a promise then we don't have a simple way of knowing when this action completed & when to re-enable the button.